add claude agent step docs (#3233)
OctopusDeploy/docs: src/pages/docs/octopus-ai/claude-agent-step/getting-started.md

Before you begin, you'll need:

- An [Anthropic API key](https://console.anthropic.com/), stored in Octopus as a **sensitive** project or library variable (for example `anthropic-api-key`). You reference it from the step as `#{anthropic-api-key}`. Storing it as a [sensitive variable](/docs/projects/variables/sensitive-variables) keeps it out of the task log.
- The [Claude Code](https://code.claude.com/docs/en/setup) CLI on the `PATH` of whatever runs the step: a worker, a deployment target, or the Octopus Server if you run the step on the server. The step launches the `claude` executable as a child process.
- If you pick the **Sandbox runtime** sandbox mode: the `srt` executable from Anthropic's [sandbox-runtime](https://code.claude.com/docs/en/sandboxing) (version 0.0.55 or later) on the `PATH`. The step checks the version and fails if it's missing or too old.

:::div{.warning}
Octopus does **not** install, download, or bootstrap the Claude Code CLI for you. If `claude` isn't on the `PATH`, the step fails when it tries to start the process. Install Claude Code on your worker or target, or bake it into your worker image, before you run the step.
:::

> 

Reviewer: We already call out that claude code is dependency above

```suggestion
```

   For a first run on a Linux worker, **Sandbox runtime** is the safest starting point. Each mode and how to configure it is covered in [Security & Compliance](/docs/octopus-ai/claude-agent-step/security-and-compliance).

   :::div{.warning}
   The **Bash sandbox** and **Sandbox runtime** modes are supported on Linux (and WSL2) workers. They aren't available on Windows workers. If you're evaluating the step on a Windows worker, use **None** and rely on isolation such as the account Tentacle runs under.
   :::

7. Select a **Permission Mode**. For your first run, select **dontAsk mode**. The step runs non-interactively, with no way to approve an action mid-run, so `dontAsk` is the standard choice: the agent may use any tool you allow and is denied everything else. The other option, Auto mode (a mode that uses a classifier model to determine whether to run a tool), is covered in [Security & Compliance](/docs/octopus-ai/claude-agent-step/security-and-compliance).
8. In **Tool Permissions**, list the tools the agent needs, one per line. For example:

   ```text
   Read
   Glob
   Bash(pwd)
   Bash(ls *)
   ```

>    Keep the list tight and add to it as you learn what the agent needs. A list that's too narrow is safer than one that's too broad. The full tool syntax is covered in [Security & Compliance](/docs/octopus-ai/claude-agent-step/security-and-compliance).

Reviewer: ```suggestion
   Start with minimal permissions and add to it as you learn what the agent needs. See [Security & Compliance](/docs/octopus-ai/claude-agent-step/security-and-compliance#tool-permissions) for more information.
```

> 9. Optionally, under **Agent Capabilities**, add **Skills** (reusable instructions) or **MCP** servers (extra tools). See [Extending the Claude Agent Step](/docs/octopus-ai/claude-agent-step/tools).

Reviewer: ```suggestion
9. Optionally, under **Agent Capabilities**, add **Skills** or **MCP** servers. See [Extending the Claude Agent Step](/docs/octopus-ai/claude-agent-step/tools).
```

10. Under **Additional Configuration Options**, set a **Turn Limit** to cap how many turns the agent can take before the step stops. One turn is a single request/response cycle with the model. The default is 10.
11. Optionally, set a **Maximum Budget** as a spend cap in USD. Leave it blank for no limit.
12. Leave **Prompt Injection Check** switched on. Before the agent runs, Octopus screens the prompt, deployment variables, MCP configuration, and skills with a fast model (`claude-haiku-4-5` by default) and blocks the step if it detects an injection attempt.
13. Save the step.

The **Run Claude Agent** step appears in your deployment process or runbook, ready to run.

## Run the step

A Run Claude Agent step runs like any other step. Run the process that contains it, then watch what the agent does in the task log.

To run the step and watch its output:

1. Create a release and deploy it, as you would for any other step.

> 2. Open the task and expand the step in the **task log**. While the step runs, the agent streams its output in real time: at the default log level you see the agent's narration and its final answer.

Reviewer: ```suggestion
2. Open the task and expand the step in the **task log**. While the step runs, the agent streams its output in real time. At the default log level you see the agent's narration and its final answer.
```

3. To also see the agent's thinking, each tool call, and the exact command Octopus ran, switch the log to verbose (or download the raw log).

A run looks roughly like this (trimmed):

```text
Running prompt-injection check against the execution context using model 'claude-haiku-4-5'.
Prompt-injection check passed: no injection detected in the execution context.
I'll help you investigate the deployment state. Let me read the deployment variables and check the working directory.
## Deployment Investigation Complete
**Deployment Summary:**
- **Project:** ClaudeIsCool
- **Environment:** Development
- **Task:** RunbookRun (ServerTasks-975)
**Working Directory Status:**
- **Writability:** CONFIRMED - Successfully created findings.txt
Claude Code usage — Duration: 15527 ms, Turns: 6
Claude Code tokens — Input: 30, Output: 1476, Cache read: 55032, Cache creation: 34098
Claude Code invocation complete.
Collecting artifacts
```

> 

Reviewer: I don't see the value in this

```suggestion
```

- The [Claude Code](https://code.claude.com/docs/en/setup) CLI on the `PATH` of whatever runs the step: a worker, a deployment target, or the Octopus Server if you run the step on the server. The step launches the `claude` executable as a child process.
- If you pick the **Sandbox runtime** sandbox mode: the `srt` executable from Anthropic's [sandbox-runtime](https://code.claude.com/docs/en/sandboxing) (version 0.0.55 or later) on the `PATH`. The step checks the version and fails if it's missing or too old.

:::div{.warning}
Octopus does **not** install, download, or bootstrap the Claude Code CLI for you. If `claude` isn't on the `PATH`, the step fails when it tries to start the process. Install Claude Code on your worker or target, or bake it into your worker image, before you run the step.
:::

To add and configure the step:

1. Open your deployment process or runbook and select **Add step**.
2. Search for `claude` and choose **Run Claude Agent** from the step library.

> <!-- SCREENSHOT: step-library-run-claude.png

Reviewer 2: Any content included in a list has to be indented... but you can also switch out of a list and use headings instead if there's lots of content within the list.

i.e.,

```markdown
## 1. Add step

Open your deployment process or runbook and select **Add step**

(screenshot)

## 2. Add the Run Claude Agent step

Search for...
```
